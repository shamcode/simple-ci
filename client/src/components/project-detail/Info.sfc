<template>
    {% import path from 'sham-ui-router/lib/params' %}
    {% import Chain from './Chain.sfc' %}

    <div class="project-info">
        <h3>{{project.name}}</h3>
        <dl>
            <dt>{{project.cwd}}</dt>
            <dd>Current work directory.</dd>
            <dd>This directory will be used when running tasks.</dd>
            <dt>
                <a :hrefto={{path("project-edit").param("id", project.id)}}>
                    Edit
                </a>
            </dt>
            <dd>Edit project name & cwd</dd>
            <dt>
                <a
                    class="project-delete-link"
                    :hrefto={{path("project-delete").param("id", project.id)}}
                >
                    Delete
                </a>
            </dt>
            <dd>Delete project</dd>
        </dl>
    </div>

    <div class="project-chains">
        <h3>Chains</h3>
        {% for chain of project.chains %}
           <Chain
                project={{project}}
                chain={{chain}}
           />
        {% endfor %}
        <a
            class="project-chain project-chain-create"
            :hrefto={{path("project-chain-create").param("id", project.id)}}
        >
            <div class="create-icon"></div>
        </a>
    </div>
</template>

<script>
    function Info( options ) {
        options( {
            [ $.project ]: { chains: [] }
        } );
    }

    export default Component( Template, Info );
</script>
