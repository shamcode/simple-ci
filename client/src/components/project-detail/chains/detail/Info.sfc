<template>
    {% import ChainForm from '../ChainForm.sfc' %}

    <div class="chain-info">
        <h3>{{chain.name}}</h3>
    </div>

    <ChainForm
            name={{chain.name}}
            command={{chain.command}}
            saveButtonText="Save"
            dataSaving={{dataSaving}}
            onSubmit={{updateProjectChain}}
    />
</template>

<script>
    function ChainInfo( options ) {
        options( {
            [ $.chain ]: {},
            [ $.dataSaving ]: false,
            [ $.updateProjectChain ]() {}
        } );
    }

    export default Component( Template, ChainInfo );
</script>
