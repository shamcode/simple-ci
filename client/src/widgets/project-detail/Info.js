import { options, DI } from 'sham-ui';
import InfoTemplate from './Info.sht';

export default class Info extends InfoTemplate {
    @options get project() { return { chains: [] }; }

    get router() {
        return DI.resolve( 'router' );
    }

    _createProjectChainURL( project ) {
        return this.router.generate( 'project-chain-create', {
            id: project.id
        } );
    }

    _routerParams( project ) {
        return {
            id: project === undefined ? null : project.id
        };
    }

    get createChainLinkNode() {
        return this.querySelector( '.project-chain-create' );
    }

    _clickToCreateNewChain( e ) {
        e.preventDefault();
        this.router.navigate( this.createChainLinkNode.getAttribute( 'href' ) );
    }
}