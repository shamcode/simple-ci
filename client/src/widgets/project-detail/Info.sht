{% import LinkTo from 'sham-ui-router/link-to' %}

<div class="project-info">
    <h3>{{project.name}}</h3>
    <dl>
        <dt>{{project.cwd}}</dt>
        <dd>Current work directory.</dd>
        <dd>This directory will be used when running tasks.</dd>
        <dt>
            <LinkTo
                    text="Edit"
                    path="project-edit"
                    params={{this._routerParams( project )}}/>
        </dt>
        <dd>Edit project name & cwd</dd>
        <dt>
            <LinkTo
                text="Delete"
                className="project-delete-link"
                path="project-delete"
                params={{this._routerParams( project )}}/>
        </dt>
        <dd>Delete project</dd>
    </dl>
</div>

<div class="project-chains">
    <h3>Chains</h3>
    {% for chain of project.chains %}
        <button
            class="project-chain"
            href="#"
        >
            {{chain.name}}
        </button>
    {% endfor %}
</div>

