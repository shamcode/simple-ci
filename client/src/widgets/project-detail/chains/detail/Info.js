import { options } from 'sham-ui';
import ChainInfoTemplate from './Info.sht';

export default class ChainInfo extends ChainInfoTemplate {
    @options get chain() { return {}; }
    @options get dataSaving() { return false; }
    @options updateProjectChain() {}
}