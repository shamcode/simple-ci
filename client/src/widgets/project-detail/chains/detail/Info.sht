{% import ChainForm from '../ChainForm' %}

<div class="chain-info">
    <h3>{{chain.name}}</h3>
</div>

<ChainForm
        name={{chain.name}}
        saveButtonText="Save"
        dataSaving={{dataSaving}}
        onSubmit={{updateProjectChain}}
/>
