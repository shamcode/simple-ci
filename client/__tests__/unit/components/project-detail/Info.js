import { DI } from 'sham-ui';
import Info from '../../../../src/components/project-detail/Info';
import * as directives from 'sham-ui-directives';
import hrefto from 'sham-ui-router/href-to';
import renderer from 'sham-ui-test-helpers';

afterEach( () => {
    DI.bind( 'store', null );
    DI.bind( 'router', null );
} );

it( 'renders correctly', () => {
    DI.bind( 'store', {
        getProjectById: jest.fn().mockReturnValue(
            Promise.resolve( {
                project: { id: 1, chains: [] }
            } )
        )
    } );
    DI.bind( 'router', {
        storage: { params: { id: 1 } },
        generate: jest.fn().mockReturnValue( '/' )
    } );

    const meta = renderer( Info, {
        directives: {
            ...directives,
            hrefto
        }
    } );
    expect( meta.toJSON() ).toMatchSnapshot();
} );
