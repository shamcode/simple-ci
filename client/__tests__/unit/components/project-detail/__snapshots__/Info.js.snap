// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "Options": Object {
    "ID": "component",
    "directives": Object {
      "EventListener": [Function],
      "UpdateOnEvent": [Function],
      "disabled": [Function],
      "hrefto": [Function],
      "onChangeUpdate": [Function],
      "onInputUpdate": [Function],
      "onclick": [Function],
      "oninput": [Function],
      "onload": [Function],
      "onresize": [Function],
      "onscroll": [Function],
      "onselect": [Function],
      "onsubmit": [Function],
      "ref": [Function],
    },
    "project": Object {
      "chains": Array [],
    },
  },
  "html": "
<div class=\\"project-info\\">
  <h3>null</h3>
  <dl>
    <dt>null</dt>
    <dd>Current work directory.</dd>
    <dd>This directory will be used when running tasks.</dd>
    <dt>
      <a href=\\"/\\"> Edit </a>
    </dt>
    <dd>Edit project name &amp; cwd</dd>
    <dt>
      <a class=\\"project-delete-link\\" href=\\"/\\"> Delete </a>
    </dt>
    <dd>Delete project</dd>
  </dl>
</div>
<div class=\\"project-chains\\">
  <h3>Chains</h3>
  <!--0-->
  <a class=\\"project-chain project-chain-create\\" href=\\"/\\">
    <div class=\\"create-icon\\"></div>
  </a>
</div>
",
}
`;
